lib/src/model/paging_state.dart

```diff
@@ -52,22 +52,22 @@ class PagingState<PageKeyType, ItemType> {
       '$itemList\u251C, error: $error, nextPageKey: $nextPageKey)';
 
   @override
-  bool operator ==(Object other) {
-    if (identical(this, other)) {
-      return true;
-    }
-    return other is PagingState &&
-        other.itemList == itemList &&
-        other.error == error &&
-        other.nextPageKey == nextPageKey;
-  }
+  bool operator ==(Object other) =>
+      identical(this, other) ||
+      (other.runtimeType == runtimeType &&
+              other is PagingState &&
+              (identical(other.error, error) || other.error == error)) &&
```

Comment:
the object default `==` behavior in dart check if the instances reference the same object. 
you can simplify `(identical(other.error, error) || other.error == error))` to `other.error == error`